src/solve/test.rs

```diff
@@ -175,7 +175,7 @@ fn prove_forall() {
         // Here, we do know that `T: Clone`, so we can.
         goal {
             forall<T> {
-                if (T: Clone) {
+                if (WellFormed(T: Clone), T: Clone) {
```

Comment:
Basically given:
```rust
trait Clone { }
impl<T> Clone for Vec<T> where T: Clone { }
```
we need to generate `(Vec<T>: Clone) :- (T: Clone), WF(T: Clone)`

because of this example:
```rust
trait Clone where Self: Marker { }
impl<T> Clone for Vec<T> where T: Clone { }
```
here we need to have `WF(T: Clone)` in order to have `Vec<T>: Clone` because `WF(T: Clone)` implies that `T` correctly implements `Marker` whereas this is not the case with the assumption `T: Clone` alone